hack/tools/boskosctl/main.go

```diff
@@ -0,0 +1,384 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package main is the main package for capv-janitor.
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/netip"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"go4.org/netipx"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"sigs.k8s.io/cluster-api-provider-vsphere/hack/tools/pkg/boskos"
+	"sigs.k8s.io/cluster-api-provider-vsphere/hack/tools/pkg/janitor"
+)
+
+var (
+	boskosHost           string
+	resourceOwner        string
+	resourceType         string
+	resourceName         string
+	vSphereUsername      string
+	vSpherePassword      string
+	vSphereServer        string
+	vSphereTLSThumbprint string
+	vSphereFolder        string
+	vSphereResourcePool  string
+)
+
+func main() {
+	log := klog.Background()
```

Comment:
Q: do the logs go to stdout or stderr?

We won't see it for acquire if it goes to stdout, right?

Reply:
I guess the answer is, it works the way we need it to :) (I tested it before)

See
> I0610 11:52:26.274161       1 main.go:135] "Creating new Boskos client" boskosHost="http://192.168.6.138:32222/" resourceOwner="pull-cluster-api-provider-vsphere-e2e-supervisor-main/1800133411847802880"
I0610 11:52:26.274286       1 main.go:193] "Acquiring resource" boskosHost="http://192.168.6.138:3[222](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-vsphere/3034/pull-cluster-api-provider-vsphere-e2e-supervisor-main/1800133411847802880#1:build-log.txt%3A222)2" resourceOwner="pull-cluster-api-provider-vsphere-e2e-supervisor-main/1800133411847802880" resourceType="vsphere-project-cluster-api-provider"
I0610 11:52:26.525741       1 main.go:198] "Acquired resource \"k8s-infra-e2e-boskos-vsphere-cluster-api-provider-vsphere-017\"" boskosHost="http://192.168.6.138:32222/" resourceOwner="pull-cluster-api-provider-vsphere-e2e-supervisor-main/1800133411847802880" resourceType="vsphere-project-cluster-api-provider"
GOBIN=/home/prow/go/src/sigs.k8s.io/cluster-api-provider-vsphere/hack/tools/bin ./hack/go-install.sh github.com/drone/envsubst/v2/cmd/envsubst envsubst v2.0.0-20210730161058-179042472c46
> https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-vsphere/3034/pull-cluster-api-provider-vsphere-e2e-supervisor-main/1800133411847802880

Otherwise it would have ended up in the env var file we source, which would have been bad